### PR TITLE
Handle Lock Push Event for Lock / Unlock

### PR DESCRIPTION
**Pull request by lenoxys**

### CODE REVIEW

**Comment by lenoxys** [comment]
hey @bropat,

tell me if you need help to review the code ? I've some people stuck on our homebridge side to have lock/unlock status right away (samemory/homebridge-eufy-security#135).

**Review by bropat — CHANGES_REQUESTED** [review]
Sorry didn't have much time to look at this.
I have now briefly looked at everything and expressed my concerns.

Can you please find out about this part and complete the PR?

Thank you!

File: src/push/types.ts [hunk 1]
```diff
@@ -65,6 +65,11 @@ export enum DoorbellPushEvent {
     ONLINE = 3107
 }
 
+export enum LockPushEvent {
```
[the remark this answers is not in the document]
Comment (lenoxys) [reply src/push/types.ts:68]: where did you get all that information ?

[the remark this answers is not in the document]
Comment (bropat) [reply src/push/types.ts:68]: By decompiling the app with for example [jadx](https://github.com/skylot/jadx)

File: src/http/device.ts [hunk 2]
```diff
@@ -1374,20 +1374,57 @@ export class Lock extends Device {
...
+                        case LockPushEvent.MECHANICAL_ANOMALY:
+                        case LockPushEvent.VIOLENT_DESTRUCTION:
+                        case LockPushEvent.MULTIPLE_ERRORS:
+                            this.updateRawProperty(CommandType.CMD_DOORLOCK_GET_STATE, { value: "5", timestamp: convertTimestampMs(message.event_time) });
+                            break;
+                        case LockPushEvent.LOW_POWE:
+                            this.updateRawProperty(CommandType.CMD_SMARTLOCK_QUERY_BATTERY_LEVEL, { value: "10", timestamp: convertTimestampMs(message.event_time) });
+                            break;
+                        case LockPushEvent.VERY_LOW_POWE:
+                            this.updateRawProperty(CommandType.CMD_SMARTLOCK_QUERY_BATTERY_LEVEL, { value: "5", timestamp: convertTimestampMs(message.event_time) });
```
Comment (bropat) [inline src/http/device.ts:1426]: Did you check or just assume?

The rest looks good.

File: src/http/device.ts [hunk 3]
```diff
@@ -1374,20 +1374,57 @@ export class Lock extends Device {
...
                             this.emit("locked", this, this.getPropertyValue(PropertyName.DeviceLocked).value as boolean);
                             break;
+                        case LockPushEvent.LOCK_MECHANICAL_ANOMALY:
+                        case LockPushEvent.MECHANICAL_ANOMALY:
+                        case LockPushEvent.VIOLENT_DESTRUCTION:
+                        case LockPushEvent.MULTIPLE_ERRORS:
+                            this.updateRawProperty(CommandType.CMD_DOORLOCK_GET_STATE, { value: "5", timestamp: convertTimestampMs(message.event_time) });
+                            break;
+                        case LockPushEvent.LOW_POWE:
+                            this.updateRawProperty(CommandType.CMD_SMARTLOCK_QUERY_BATTERY_LEVEL, { value: "10", timestamp: convertTimestampMs(message.event_time) });
```
Comment (bropat) [inline src/http/device.ts:1423]: CommandType.CMD_SMARTLOCK_QUERY_BATTERY_LEVEL corresponds to the battery value in %.

Are you sure that this message is always the same as 10% battery?

File: src/http/device.ts [hunk 4]
```diff
@@ -1374,20 +1374,57 @@ export class Lock extends Device {
...
+                            break;
+                        case LockPushEvent.VERY_LOW_POWE:
```
Comment (bropat) [inline src/http/device.ts:1425]: CommandType.CMD_SMARTLOCK_QUERY_BATTERY_LEVEL corresponds to the battery value in %.

Are you sure that this message is always the same as 5% battery?

File: src/http/device.ts (hunk 2 above)
Comment (lenoxys) [reply src/http/device.ts:1426]: assume I don't have a lock at home

File: src/http/device.ts (hunk 3 above)
Comment (lenoxys) [reply src/http/device.ts:1423]: Not sure but assuming. Looks like the lock is not sending back the battery level but a low or very low battery notification

File: src/http/device.ts (hunk 4 above)
Comment (lenoxys) [reply src/http/device.ts:1425]: Not sure but assuming. Looks like the lock is not sending back the battery level but a low or very low battery notification

File: src/http/device.ts (hunk 3 above)
Comment (bropat) [reply src/http/device.ts:1423]: I would suggest that we comment out this part and as soon as we have a more precise idea of how we need to handle these events we will take up the topic again.

File: src/http/device.ts (hunk 4 above)
Comment (bropat) [reply src/http/device.ts:1425]: I would suggest that we comment out this part and as soon as we have a more precise idea of how we need to handle these events we will take up the topic again.

File: src/http/types.ts [hunk 5]
```diff
@@ -897,13 +897,23 @@ export const DeviceBasicLockStatusProperty: PropertyMetadataNumeric = {
...
+        8: "APP_LOCK",
+        9: "AUTO_LOCK",
+        10: "LOW_POWER",
+        11: "VERY_LOW_POWER",
+        12: "MULTIPLE_ERRORS",
+        13: "LOCK_OFFLINE",
+        14: "MECHANICAL_ANOMALY",
+        15: "VIOLENT_DESTRUCTION",
+        16: "PW_LOCK",
+        17: "FINGER_LOCK",
```
Comment (bropat) [inline src/http/types.ts:916]: Please revert this change

**Review by bropat — CHANGES_REQUESTED** [review]

File: src/http/types.ts (hunk 5 above)
Comment (bropat) [reply src/http/types.ts:916]: After you reverted this part i will merge, please don't change anything after that.
